pkg/sql/alter_table.go

```diff
@@ -1964,10 +1946,22 @@ func handleTTLStorageParamChange(
 		}
 	}
 
-	if dropTTLMutation {
+	// Adding TTL requires adding the TTL job before adding the TTL fields.
+	// Removing TTL requires removing the TTL job before removing the TTL fields.
+	var direction descpb.DescriptorMutation_Direction
+	switch {
```

Comment:
Consolidated these cases into a switch to make it clear only one case will happen.